pkg/search/postgres/store_cache.go

```diff
@@ -153,10 +153,18 @@ func (c *cachedStore[T, PT]) Delete(ctx context.Context, id string) error {
 		obj, found := c.cache[id]
 		return obj, found
 	})
-	if !found {
+	if !c.isWriteAllowed(ctx, obj) {
+		// Special case: the query generator for global scoped resources currently returns sac.ErrResourceAccessDenied
+		// when write is not allowed. The cached store needs to respect that behavior for the time being.
+		// Ultimately, we have to convey on the behavior: either _always_ return sac.ErrResourceAccessDenied or return
+		// nil.
+		// TODO(ROX-XXXXX): Once the behavior is fixed, remove this special casing.
```

Comment:
Once JIRA allows me to login again, I'll create the ticket and fix this reference.